service/service.go

```diff
@@ -49,11 +50,17 @@ func (svc *Service) Init(ctx context.Context, cfg *config.Config, serviceList *E
 	// Get health client for api router
 	svc.routerHealthClient = serviceList.GetHealthClient("api-router", cfg.APIRouterURL)
 
+	dimensionClient, err := dimension.NewWithHealthClient(svc.routerHealthClient)
+	if err != nil {
+		return fmt.Errorf("failed to create dimensions API client: %w", err)
+	}
+
 	// Initialise clients
 	clients := routes.Clients{
-		Render:  render.NewWithDefaultClient(assets.Asset, assets.AssetNames, cfg.PatternLibraryAssetsPath, cfg.SiteDomain),
-		Filter:  filter.NewWithHealthClient(svc.routerHealthClient),
-		Dataset: dataset.NewWithHealthClient(svc.routerHealthClient),
+		Render:    render.NewWithDefaultClient(assets.Asset, assets.AssetNames, cfg.PatternLibraryAssetsPath, cfg.SiteDomain),
+		Filter:    filter.NewWithHealthClient(svc.routerHealthClient),
+		Dataset:   dataset.NewWithHealthClient(svc.routerHealthClient),
+		Dimension: dimensionClient,
```

Comment:
```suggestion
		Dimension: dimension.NewWithHealthClient(svc.routerHealthClient),
```

Reply:
Error handling should be handled here ^

Reply:
Each property on `routes.Clients` is a valid/constructed API client. Since the Dimensions API constructor can fail (it uses `url.URL`, which the others don't), I don't think I can avoid instantiating it above and passing it here like this.